test/ref/roialign.cpp

```diff
@@ -84,114 +84,164 @@ TEST_CASE(roialign_out_of_bound_test)
     }
 }
 
+auto create_program(const std::string& trans_mode = "half_pixel",
```

Comment:
Notes on ref tests:  added cases with all 4 combinations of trans_mode and pooling_mode and split them apart into separate named cases.  The modified `create_program()` has a reshaped input with multiple channels, multiple layers, an ROI list that doesn't match 1-1 with the layers, non-unity scale and sampling ratio, one negative data value (any float value is legal) and an ROI that goes out of bounds (also legal).  Also, the output height and width are no longer equal, which masked errors in the original implementation.

Need to debug the max pooling cases!